src/apify/storage_clients/_apify/_utils.py

```diff
@@ -1,117 +1,170 @@
-        if isinstance(record, dict) and 'value' in record:
-            record = record['value']
-
-        # Extract the actual data from the KVS record
-        if isinstance(record, dict) and alias_key in record:
-            storage_id = record[alias_key]
-            return str(storage_id)
 
-    except Exception as exc:
-        # If there's any error accessing the record, treat it as not found.
-        logger.warning(f'Error accessing alias mapping for {alias}: {exc}')
+class Alias:
+    """Class for handling aliases.
 
-    return None
-
-
-async def store_alias_mapping(
-    alias: str,
-    storage_type: Literal['dataset', 'kvs', 'rq'],
-    storage_id: str,
-    configuration: Configuration,
-) -> None:
-    """Store a mapping from alias to storage ID in the default key-value store.
-
-    Args:
-        alias: The alias to store.
-        storage_type: Type of storage ('dataset', 'key_value_store', or 'request_queue').
-        storage_id: The storage ID to map the alias to.
-        configuration: The configuration object containing API credentials.
+    The purpose of this is class is to ensure that alias storages are created with correct id. This is achieved by using
+    default kvs as a storage for global mapping of aliases to storage ids. Same mapping is also kept in memory to avoid
+    unnecessary calls to API and also have limited support of alias storages when not running on Apify platform. When on
+     Apify platform, the storages created with alias are accessible by the same alias even after migration or reboot.
     """
-    default_kvs_client = await _get_default_kvs_client(configuration)
-
-    # Create the dictionary key for this alias.
-    alias_key = f'alias-{storage_type}-{alias}'
-
-    try:
-        record = await default_kvs_client.get_record(_ALIAS_MAPPING_KEY)
-
-        # get_record can return {key: ..., value: ..., content_type: ...}
-        if isinstance(record, dict) and 'value' in record:
-            record = record['value']
 
-        # Update or create the record with the new alias mapping
-        if isinstance(record, dict):
-            record[alias_key] = storage_id
-        else:
-            record = {alias_key: storage_id}
-
-        # Store the mapping back in the KVS.
-        await default_kvs_client.set_record(_ALIAS_MAPPING_KEY, record)
-    except Exception as exc:
-        logger.warning(f'Error accessing alias mapping for {alias}: {exc}')
-
-
-async def _get_default_kvs_client(configuration: Configuration) -> KeyValueStoreClientAsync:
+    _alias_map: ClassVar[dict[str, str]] = {}
+    """Map containing pre-existing alias storages and their ids. Global for all instances."""
+    _alias_init_lock: Lock | None = None
+    """Lock for creating alias storages. Only one alias storage can be created at the time. Global for all instances."""
+
+    ALIAS_STORAGE_KEY_SEPARATOR = ','
+    ALIAS_MAPPING_KEY = '__STORAGE_ALIASES_MAPPING'
```

Comment:
These could be private class attrs.

```suggestion
    _ALIAS_STORAGE_KEY_SEPARATOR = ','
    _ALIAS_MAPPING_KEY = '__STORAGE_ALIASES_MAPPING'
```